proxy/internal/modules/store/mysql.go

```diff
@@ -0,0 +1,496 @@
+package store
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"oss.indeed.com/go/modprox/pkg/coordinates"
+	"oss.indeed.com/go/modprox/pkg/loggy"
+	"oss.indeed.com/go/modprox/pkg/metrics/stats"
+	"oss.indeed.com/go/modprox/pkg/repository"
+)
+
+type mysqlStore struct {
+	emitter    stats.Sender
+	db         *sql.DB
+	statements statements
+	log        loggy.Logger
+}
+
+var _ ZipStore = (*mysqlStore)(nil)
+var _ Index = (*mysqlStore)(nil)
+
+const dbTimeout = 10 * time.Second
+
+func (m *mysqlStore) PutZip(mod coordinates.Module, blob repository.Blob) error {
+	start := time.Now()
+	err := m.insertModulesContents(mod, blob)
+	if err != nil {
+		m.emitter.Count("db-put-zip-failure", 1)
+		return err
+	}
+
+	m.emitter.GaugeMS("db-put-zip-elapsed-ms", start)
+	return nil
+}
+
+func (m *mysqlStore) GetZip(mod coordinates.Module) (repository.Blob, error) {
+	m.log.Tracef("retrieving module %s", mod)
+
+	start := time.Now()
+	blob, err := m.getModuleContents(mod)
+	if err != nil {
+		return nil, err
```

Comment:
log an error

Reply:
shouldn't the caller be logging the error?

Reply:
this specific bit of code was removed, but there are similar examples in the same file. I didn't add any logging because I think the caller should be logging it.